### PR TITLE
PWN-7116 - mapper for loaded swap settings list

### PR DESCRIPTION
**Pull request by NikolaiEmelianov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7116

## Description of Work

mapper for loaded swap settings list

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateManager.kt [hunk 1]
```diff
@@ -23,7 +23,7 @@ class SwapStateManager(
 ) : CoroutineScope {
 
     companion object {
-        const val DEFAULT_ACTIVE_ROUTE_ORDINAL = 1
+        const val DEFAULT_ACTIVE_ROUTE_ORDINAL = 0
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateManager.kt:26]: i think if it's 0 now - it's DEFAULT_ACTIVE_ROUTE_INDEX

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    private val commonMapper: SwapCommonSettingsMapper
+) {
+
+    fun mapForLoadingTransactionState(
+        slippage: Double,
+        routes: List<JupiterSwapRoute>,
+        activeRoute: Int,
+        jupiterTokens: List<JupiterSwapToken>,
+        tokenB: SwapTokenModel,
+    ): List<AnyCellItem> = mapList(slippage, routes, activeRoute, jupiterTokens, null, tokenB)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt:34]: named args

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+    fun mapForSwapLoadedState(
+        slippage: Double,
+        routes: List<JupiterSwapRoute>,
+        activeRoute: Int,
+        jupiterTokens: List<JupiterSwapToken>,
+        tokenBAmount: BigDecimal?,
+        tokenB: SwapTokenModel,
+    ): List<AnyCellItem> = mapList(slippage, routes, activeRoute, jupiterTokens, tokenBAmount, tokenB)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt:43]: named args

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+    private fun mapList(
+        slippage: Double,
+        routes: List<JupiterSwapRoute>,
+        activeRoute: Int,
+        jupiterTokens: List<JupiterSwapToken>,
+        tokenBAmount: BigDecimal?,
+        tokenB: SwapTokenModel,
+    ): List<AnyCellItem> = mutableListOf<AnyCellItem>().apply {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt:52]: build list

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+        addRouteCell(routes, activeRoute, jupiterTokens)
+        this += commonMapper.getNetworkFeeCell()
+        addAccountFeeCell()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt:55]: i think it's better to have add in one way

so everything is
this +=
this +=
this +=

or everything is 

addLiquidityFeeCell
addEstimatedFeeCell
addMinimumReceivedCell

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+                    iconTint = R.color.icons_mountain,
+                )
+            ),
+            payload = SwapSettingsPayload.ROUTE,
+            styleType = FinanceBlockStyle.BASE_CELL,
+        )
+    }
+
+    private fun formatRouteString(route: JupiterSwapRoute?, jupiterTokens: List<JupiterSwapToken>): String {
+        if (route == null) return ""
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt:96]: emptyString

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+        var result = ""
+        route.marketInfos.forEachIndexed { index, marketInfo ->
+            result = if (index != route.marketInfos.lastIndex) {
+                result.plus(findTokenSymbolByMint(marketInfo.inputMint, jupiterTokens)).plus("→")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt:100]: maybe we can rewrite this with buildString ?

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+        tokenBAmount: BigDecimal?,
+        tokenB: SwapTokenModel
+    ) {
+        val secondLineText = if (tokenBAmount == null) {
+            TextViewCellModel.Skeleton(
+                skeleton = leftSubtitleSkeleton()
+            )
+        } else {
+            TextViewCellModel.Raw(
+                text = TextContainer(tokenBAmount.multiply(slippage.toBigDecimal()).formatToken(tokenB.decimals))
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt:119]: extract tokenBAmount.multiply(slippage.toBigDecimal()).formatToken(tokenB.decimals) to variable to split meaning

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+                )
+            ),
+            payload = SwapSettingsPayload.CREATION_FEE,
+            styleType = FinanceBlockStyle.BASE_CELL,
+        )
+    }
+
+    private fun MutableList<AnyCellItem>.addLiquidityFeeCell(
+        routes: List<JupiterSwapRoute>,
+        activeRoute: Int,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt:166]: activeRouteIndex

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+        jupiterTokens: List<JupiterSwapToken>
+    ) {
+        val route = routes.getOrNull(activeRoute)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt:169]: route  -> activeRoute

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt [hunk 11]
```diff
@@ -0,0 +1,242 @@
...
+
+    private fun rightSideSkeleton(): SkeletonCellModel {
+        return SkeletonCellModel(
+            height = 16.toPx(),
+            width = 52.toPx(),
+            radius = 4f.toPx(),
+        )
+    }
+
+    private fun findTokenSymbolByMint(mint: Base58String, jupiterTokens: List<JupiterSwapToken>): String {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt:239]: maybe make it as extension for List<JupiterSwapToken>?

**Review by gslevinkov — APPROVED** [review]

**Review by Davran331122 — APPROVED** [review]

**Pull request merged**